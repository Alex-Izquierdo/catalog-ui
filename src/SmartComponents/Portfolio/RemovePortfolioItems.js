import React from 'react';
import propTypes from 'prop-types';
import { Button, Title } from '@patternfly/react-core';
import './removeportfolioitems.scss';
import { Link } from 'react-router-dom';

const RemovePortfolioItems = ({ portfolioRoute, onRemove }) => (
  <div className="remove">
    <Title size={ '2x1' }>Remove Products</Title>
    <div className="buttons_right">
      <Link to={ portfolioRoute }>
        <Button variant="link" aria-label="Cancel removing portfolio items">Cancel</Button>
      </Link>
      <Button variant="primary" aria-label="Remove selected portfolio items" onClick={ onRemove }>Remove</Button>
    </div>
  </div>
);

RemovePortfolioItems.propTypes = {
  onCancel: propTypes.func,
  onRemove: propTypes.func,
  portfolioRoute: propTypes.string
};

export default RemovePortfolioItems;
