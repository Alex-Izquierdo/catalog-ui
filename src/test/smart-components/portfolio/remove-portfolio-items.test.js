import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import RemovePortfolioItems from '../../../SmartComponents/Portfolio/RemovePortfolioItems';

describe('<RemovePortfolioItems />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      filterValue: '',
      onFilterChange: jest.fn(),
      portfolioRoute: '/',
      portfolioName: 'Foo'
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<RemovePortfolioItems { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
