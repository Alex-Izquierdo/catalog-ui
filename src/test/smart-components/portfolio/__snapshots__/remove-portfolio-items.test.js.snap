// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RemovePortfolioItems /> should render correctly 1`] = `
<TopToolbar
  paddingBottom={true}
>
  <TopToolbarTitle
    title="Remove products: Foo"
  />
  <Level
    className=""
    gutter={null}
  >
    <Toolbar
      className={null}
    >
      <ToolbarGroup
        className={null}
      >
        <ToolbarItem
          className={null}
        >
          <FilterToolbarItem
            onFilterChange={[MockFunction]}
            placeholder="Filter by name..."
            searchValue=""
          />
        </ToolbarItem>
      </ToolbarGroup>
      <ToolbarGroup
        className={null}
      >
        <ToolbarItem
          className={null}
        >
          <Link
            to="/"
          >
            <Button
              aria-label="Cancel removing portfolio items"
              className=""
              component="button"
              isActive={false}
              isBlock={false}
              isDisabled={false}
              isFocus={false}
              isHover={false}
              isInline={false}
              type="button"
              variant="link"
            >
              Cancel
            </Button>
          </Link>
        </ToolbarItem>
      </ToolbarGroup>
      <ToolbarGroup
        className={null}
      >
        <ToolbarItem
          className={null}
        >
          <Button
            aria-label="Remove selected portfolio items"
            className=""
            component="button"
            isActive={false}
            isBlock={false}
            isDisabled={false}
            isFocus={false}
            isHover={false}
            isInline={false}
            type="button"
            variant="danger"
          >
            Remove
          </Button>
        </ToolbarItem>
      </ToolbarGroup>
    </Toolbar>
  </Level>
</TopToolbar>
`;
